### PR TITLE
Remove some checks of `dpl::underlying_type`

### PR DESCRIPTION
**Pull request by SergeyKopienko**
Fix compile error under `icpx` compiler:

```C++
icpx -DTBB_USE_GLIBCXX_VERSION=80401 -D_PSTL_TEST_SUCCESSFUL_KEYWORD=1 -I/test -I/include -isystem /tbb/latest/include -DTEST_LONG_RUN=1 -O2 -DNDEBUG -fno-fast-math -fopenmp-simd -fsycl -fno-sycl-unnamed-lambda -std=c++20 -o CMakeFiles/underlying_type.pass.dir/xpu_api/utilities/meta/meta.trans/meta.trans.other/underlying_type.pass.cpp.o -c /test/xpu_api/utilities/meta/meta.trans/meta.trans.other/underlying_type.pass.cpp
In file included from test/xpu_api/utilities/meta/meta.trans/meta.trans.other/underlying_type.pass.cpp:18:
In file included from include/oneapi/dpl/type_traits:20:
/usr/lib/gcc/x86_64-redhat-linux/8/../../../../include/c++/8/type_traits:2002:15: error: only enumeration types have underlying types
 2002 |       typedef __underlying_type(_Tp) type;
      |               ^
test/xpu_api/utilities/meta/meta.trans/meta.trans.other/has_type_member.h:27:48: note: in instantiation of template class 'std::underlying_type<void>' requested here
   27 | struct has_type_member<T, dpl::void_t<typename T::type>> : dpl::true_type
      |                                                ^
test/xpu_api/utilities/meta/meta.trans/meta.trans.other/underlying_type.pass.cpp:102:28: note: during template argument deduction for class template partial specialization 'has_type_member<T, dpl::void_t<typename T::type>>' [with T = std::underlying_type<void>]
  102 |             static_assert(!has_underlying_type_member<void>::value);
```

```C++
/test/xpu_api/utilities/meta/meta.trans/meta.trans.other/underlying_type.pass.cpp:114:28: note: in instantiation of template class 'has_type_member<std::underlying_type<std::nullptr_t>>' requested here
In file included from /test/xpu_api/utilities/meta/meta.trans/meta.trans.other/underlying_type.pass.cpp:18:
In file included from /include/oneapi/dpl/type_traits:20:
/usr/lib/gcc/x86_64-redhat-linux/8/../../../../include/c++/8/type_traits:2002:15: error: only enumeration types have underlying types
 2002 |       typedef __underlying_type(_Tp) type;
      |               ^
/test/xpu_api/utilities/meta/meta.trans/meta.trans.other/has_type_member.h:27:48: note: in instantiation of template class 'std::underlying_type<double>' requested here
```

### CODE REVIEW

File: test/xpu_api/utilities/meta/meta.trans/meta.trans.other/underlying_type.pass.cpp [hunk 1]
```diff
@@ -99,6 +99,7 @@ kernel_test1(sycl::queue& deviceQueue)
 
             // Until C++20 the behaviour of std::underlying_type<T> is undefined when T is not enum
 #if TEST_STD_VER >= 20
+#    if !_PSTL_ICC_TEST_UNDERLYING_TYPE_BROKEN
```
Comment (kboyarinov) [inline test/xpu_api/utilities/meta/meta.trans/meta.trans.other/underlying_type.pass.cpp:102]:
```suggestion
#if TEST_STD_VER >= 20 && !_PSTL_ICC_TEST_UNDERLYING_TYPE_BROKEN
```

File: test/xpu_api/utilities/meta/meta.trans/meta.trans.other/underlying_type.pass.cpp [hunk 2]
```diff
@@ -127,18 +130,21 @@ kernel_test2(sycl::queue& deviceQueue)
...
     sycl::queue deviceQueue = TestUtils::get_test_queue();
     kernel_test1(deviceQueue);
 
+#if !_PSTL_ICC_TEST_UNDERLYING_TYPE_BROKEN
     const auto device = deviceQueue.get_device();
     if (TestUtils::has_type_support<double>(device))
     {
         kernel_test2(deviceQueue);
     }
+#endif
```
Comment (kboyarinov) [inline test/xpu_api/utilities/meta/meta.trans/meta.trans.other/underlying_type.pass.cpp:147]: I propose to wrap the entire code block with `TEST_STD_VER >= 20 && !_PSTL_ICC_TEST_UNDERLYING_TYPE_BROKEN`

We can skip making copy of the device as well in this case.

File: test/xpu_api/utilities/meta/meta.trans/meta.trans.other/underlying_type.pass.cpp [hunk 3]
```diff
@@ -112,11 +113,13 @@ kernel_test1(sycl::queue& deviceQueue)
             static_assert(!has_underlying_type_member<int&&>::value);
             static_assert(!has_underlying_type_member<int*>::value);
             static_assert(!has_underlying_type_member<dpl::nullptr_t>::value);
+#   endif
 #endif
         });
     });
 }
 
+#if !_PSTL_ICC_TEST_UNDERLYING_TYPE_BROKEN
```
Comment (kboyarinov) [inline test/xpu_api/utilities/meta/meta.trans/meta.trans.other/underlying_type.pass.cpp:122]: Let's also merge two checks for entire function: `TEST_STD_VER >= 20 && !_PSTL_ICC_TEST_UNDERLYING_TYPE_BROKEN`

Comment (SergeyKopienko) [reply test/xpu_api/utilities/meta/meta.trans/meta.trans.other/underlying_type.pass.cpp]: Fixed

File: test/xpu_api/utilities/meta/meta.trans/meta.trans.other/underlying_type.pass.cpp (hunk 2 above)
Comment (SergeyKopienko) [reply test/xpu_api/utilities/meta/meta.trans/meta.trans.other/underlying_type.pass.cpp]: Fixed

**Review by kboyarinov — APPROVED** [review]
LGTM

**Pull request merged**